### PR TITLE
Update benchmarks

### PR DESCRIPTION
**Pull request by DonggeLiu**
1. Allow generating benchmarks with multiple oracles, prioritizing by oracle order.
2. Update benchmarks using the new oracles: `optimal-targets`, `all-public-candidates`.

@DavidKorczynski It appears that `optimal-targets` does not work well with many benchmarks.
E.g., empty list for `libxml2`, 'Project not in the database' for `avahi`. Is this expected?
```
https://introspector.oss-fuzz.com/api/optimal-targets?project=libxml2&exclude-static-functions=True&only-referenced-functions=False&only-with-header-file-declaration=True
{'functions': [], 'result': 'success'}
https://introspector.oss-fuzz.com/api/optimal-targets?project=avahi&exclude-static-functions=True&only-referenced-functions=False&only-with-header-file-declaration=True
{'msg': 'Project not in the database', 'result': 'error'}
```

### CODE REVIEW

File: data_prep/introspector.py [hunk 1]
```diff
@@ -747,7 +755,8 @@ def _parse_arguments() -> argparse.Namespace:
   parser.add_argument('-t',
                       '--target-oracle',
                       type=str,
-                      default='far-reach-low-coverage',
+                      nargs='+',
+                      default=['optimal-targets', 'all-public-candidates'],
```
Comment (DavidKorczynski) [inline data_prep/introspector.py:759]: I'd keep `far-reach-low-coverage` as the primary benchmark -- there will still be improvements with respect to build rate from the use of `exclude-static-functions=True&only-referenced-functions=False&only-with-header-file-declaration=True`

`optimal-targets` is a bit more intricate and has far fewer functions per project. For some there may be no due to additional filtering from `exclude-static-functions=True&only-referenced-functions=False&only-with-header-file-declaration=True`

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: data_prep/introspector.py (hunk 1 above)
Comment (DonggeLiu) [reply data_prep/introspector.py:759]: I see, I will use `['optimal-targets', 'far-reach-low-coverage']` then.

> all-public-candidates is primarily a JVM oracle. 

Could you please reflect that in the name of `INTROSPECTOR_ORACLE_ALL_CANDIDATES`, like `INTROSPECTOR_ALL_JVM_SOURCE_PATH`?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
>   2. Update benchmarks using the new oracles: optimal-targets, all-public-candidates.

I understood it as you'd like to also add new benchmarks? At the moment it's only removing `all`

**Comment by DonggeLiu** [comment]
> > 2. Update benchmarks using the new oracles: optimal-targets, all-public-candidates.
> 
> I understood it as you'd like to also add new benchmarks? At the moment it's only removing `all`

Yep, I need to confirm which oracles to use before generate new benchmarks : )

**Comment by DavidKorczynski** [comment]
> Yep, I need to confirm which oracles to use before generate new benchmarks : )

Do: `far-reach-but-low-coverage`, `optimal-targets`, `easy-params-far-reach`

**Comment by DonggeLiu** [comment]
Looking OK:
https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-07-05-431-dg-comparison/index.html

**Review by DavidKorczynski — APPROVED** [review]

**Comment by DonggeLiu** [comment]
Looking good:
https://llm-exp.oss-fuzz.com/Result-reports/scheduled/2024-07-06-weekly-all/

Will select a new comparison set for daily improvement/regression monitoring.

**Comment by DonggeLiu** [comment]
The new `comparison` set is selected from the [07-06 experiment](https://llm-exp.oss-fuzz.com/Result-reports/scheduled/2024-07-06-weekly-all/), containing benchmarks with:
1. Top performance
2. Mid/Low performing one
3. 0 build rate (majority).

**Pull request merged**